src/JavaCodeGenerator.cs

```diff
@@ -1213,13 +1214,19 @@ private static RestAPIMethod ParseRestAPIMethod(AutoRestMethod autoRestMethod, J
 
                 bool parameterSkipUrlEncodingExtension = GetExtensionBool(autoRestParameter.Extensions, SwaggerExtensions.SkipUrlEncodingExtension);
 
+                string parameterHeaderCollectionPrefix = GetExtensionString(autoRestParameter.Extensions, SwaggerExtensions.HeaderCollectionPrefix);
+                if (!string.IsNullOrEmpty(parameterHeaderCollectionPrefix))
+                {
+                    parameterType = new MapType(ClassType.String);
```

Comment:
Does this constructor indicate basically that AutoRest only supports strings as map keys anyway?

Reply:
Correct.